python/mlx/nn/layers/normalization.py

```diff
@@ -4,6 +4,80 @@
 from mlx.nn.layers.base import Module
 
 
+class InstanceNorm(Module):
+    r"""Applies instance normalization [1] on the inputs.
+
+    Computes
+
+    .. math::
+
+        y = \frac{x - \mathrm{E}[x]}{ \sqrt{\mathrm{Var}[x] + \epsilon}} * \gamma + \beta,
+
+    where :math:`\gamma` and :math:`\beta` are learned per feature dimension
+    parameters initialized at 1 and 0 respectively. Both are of size num_features,
+    if :attr:`affine` is ``True``.
+
+    [1]: https://arxiv.org/abs/1607.08022
+
+    Args:
+        num_features: number of features of the input
+        eps: a value added to the denominator for numerical stability. Default: 1e-5
+        momentum: the value used for the running_mean and running_var computation. Default: 0.1
+        affine: a boolean value that when set to ``True``, this module has
+            learnable affine parameters, initialized the same way as done for batch normalization.
+            Default: ``False``.
+    Shape:
+        - Input: :math:`(N, C, L)` or :math:`(N, C, H, W)` or :math:`(N, C, D, H, W)`
+        - Output: :math:`(N, C, L)` or :math:`(N, C, H, W)` or :math:`(N, C, D, H, W)` (same shape as input)
+    """
+
+    def __init__(
+        self,
+        num_features: int,
+        eps: float = 1e-5,
+        affine: bool = False,
+    ):
+        super().__init__()
+        if affine:
+            self.weight = mx.ones((num_features,))
+            self.bias = mx.zeros((num_features,))
+        self.eps = eps
+        self.affine = affine
+        self.param_shapes = {
+            3: (1, num_features, 1),  # input shape: (B, C, L)
+            4: (1, num_features, 1, 1),  # input shape: (B, C, H, W)
+            5: (1, num_features, 1, 1, 1),  # input shape: (B, C, D, H, W)
+        }
```

Comment:
This should be changed to have channels last which is a convention we follow. It also simplifies stuff because then the weight and bias broadcast with the input so you can just get rid of the `param_shapes` map.